.github/workflows/beam_PreCommit_Spotless.yml

```diff
@@ -0,0 +1,104 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+name: PreCommit Spotless
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths:
+      - 'buildSrc/**'
+      - 'sdks/java/**'
+      - 'runners/**'
+      - 'examples/java/**'
+      - 'examples/kotlin/**'
+      - '.test-infra/jenkins/'
+      - '.github/workflows/beam_PreCommit_Spotless.yml'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths:
+      - 'buildSrc/**'
+      - 'sdks/java/**'
+      - 'runners/**'
+      - 'examples/java/**'
+      - 'examples/kotlin/**'
+      - '.test-infra/jenkins/'
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  beam_PreCommit_Spotless:
+    name: beam_PreCommit_Spotless (Run Spotless PreCommit)
+    timeout-minutes: 120
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Spotless PreCommit'
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.pull_request.head.sha}}
+      - name: Rerun on comment
+        if: github.event.comment.body == 'Run Spotless PreCommit'
+        uses: ./.github/actions/rerun-job-action
+        with:
+          pull_request_url: ${{ github.event.issue.pull_request.url }}
+          github_repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ github.job }}
+          github_current_run_id: ${{ github.run_id }}
+      - name: Install Java
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+          cache: 'gradle'
+          check-latest: true
+      - name: run Spotless PreCommit script
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:java:core:jmh:spotlessCheck :sdks:java:core:jmh:checkStyleMain :sdks:java:core:jmh:checkStyleTest
```

Comment:
thi is not correct. It just run against `:sdks:java:core:jmh` module. Should run on all java modules